mono/metadata/reflection.c

```diff
@@ -1228,6 +1228,15 @@ mono_custom_attrs_free (MonoCustomAttrInfo *ainfo)
 		g_free (ainfo);
 }
 
+void
+mono_custom_attrs_free_cached (MonoCustomAttrInfo *ainfo)
+{
+	if (ainfo) {
+		g_assert (ainfo->cached);
```

Comment:
Are we sure `ainfo->cached` is always true? The `mono_custom_attrs_free` function will do nothing when `ainfo->cached` is true, which suggests that it might be false in some cases.

Reply:
Yes, if we successfully insert a non-NULL value we set the cached flag: https://github.com/Unity-Technologies/mono/blob/065c52b430e6d71731d0f45a441cf48a88f62cf1/mono/metadata/reflection.c#L8421

Reply:
The problem is `mono_custom_attrs_free` is called from many clients with the purpose to never free cached items. In this case, we explicitly want to clean up the hash table of cached values.

Reply:
Ok, that makes sense, thanks!